0016-pre-shared-key/0016-pre-shared-key.md

```diff
@@ -0,0 +1,57 @@
+# Pre Shared Key
+
+## Data Format
+
+The PSK details are divided into two HTTP-style request objects. Each of the
+requests starts with `PSK/1.0` and then `INSECURE` or `SECURE`.  The requests
+are made of a list of headers, followed by data. For simplicity, duplicate
+headers are not allowed. 
+
+### Outer Request
+
+The outer request is formatted as below:
+
+```http
+PSK/1.0 INSECURE
+Key-Id: fpwpAhlN588
+Unsafe-Header: data_everyone_can_see
```

Comment:
We should note that implementations SHOULD (or maybe MUST) prefix all headers sent in the clear with `Unsafe-`